rimage/transform/parallel_projection.go

```diff
@@ -95,5 +97,162 @@ func (pp *ParallelProjection) PointCloudToRGBD(cloud pointcloud.PointCloud) (*ri
 
 // ImagePointTo3DPoint takes the 2D pixel point and assumes that it represents the X,Y coordinate in mm as well.
 func (pp *ParallelProjection) ImagePointTo3DPoint(pt image.Point, d rimage.Depth) (r3.Vector, error) {
-	return r3.Vector{float64(pt.X), float64(pt.Y), float64(d)}, nil
+	return r3.Vector{X: float64(pt.X), Y: float64(pt.Y), Z: float64(d)}, nil
+}
+
+// ParallelProjectionOntoXZWithRobotMarker allows the creation of a 2D projection of a pointcloud and robot
+// position onto the XZ plane.
+type ParallelProjectionOntoXZWithRobotMarker struct {
+	robotPose *spatialmath.Pose
+}
+
+const (
+	sigmaLevel       = 7
+	imageHeight      = 1080
+	imageWidth       = 1070
+	missThreshold    = 0.44
+	hitThreshold     = 0.53
```

Comment:
What does it mean if a value is between the hit & miss threshold & how will parallel projection behave in that situation?

Reply:
Added descriptions for these. As I might have said in an earlier comment these are values and concepts taken from cartographer but will need to be fine tuned upon integrations.

I believe cartographer assigns a hit or miss color to points coloring them (in our cartographer's case green or yellow) respectively. We may choose to change that color scheme once we see what the data is like (waiting on [RSDK-1074](https://viam.atlassian.net/browse/RSDK-1074))

Reply:
Maybe another silly question but why do we have cartogropher specific concepts in the generic XZ image generation code? 

I thought the goal was to have this code be generic across any slam algo?